Formula/g/glib.rb

```diff
@@ -54,6 +54,7 @@ class Glib < Formula
 
   # replace several hardcoded paths with homebrew counterparts
   patch do
+    # TEST: reverted path: does this actually build?
```

Comment:
Not sure why you're skeptical that the breakage is related to your patch -- here's a run from a few weeks ago where `glib` was rebuilt and those dependents were testing successfully: https://github.com/Homebrew/homebrew-core/actions/runs/11435239157/job/31810997342?pr=195052

Or, for a more recent test, `graph-tool` was rebuilt three days ago: https://github.com/Homebrew/homebrew-core/actions/runs/11624040843/job/32372066536?pr=193993

The regression is definitely due to the patch changes, unfortunately.

Reply:
It's building only libraries (in)directly dependent on GLib?

It's not that I'm skeptical. I cannot see any obvious reason why this build fails. I'm pretty sure the breaking libraries do not depend on `girepository`, so I wonder why they would break at all.

I figured I'll start from a clean branch add add the changes incrementally.

Reply:
Thanks. I'll try to reproduce the issues locally.

Reply:
I created a little test script:

```python
#!/opt/homebrew/bin/python3.13
import gi
gi.require_version("GIRepository", "2.0")
from gi.repository import GIRepository

repo=GIRepository.Repository()
print("Typelib search path", repo.get_search_path())
```

I installed `pygobject3` and the script works.

When I `brew reinstall -s glib` using homebrew-core's master branch, and run the script I get the error you describe:

```
** (process:38422): WARNING **: 12:06:09.911: Failed to load shared library '/private/tmp/glib-20241104-31967-
kl1vyd/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' referenced by the typelib: dlopen(/private/tmp/
glib-20241104-31967-kl1vyd/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib, 0x0009): tried: '/private/tmp/
glib-20241104-31967-kl1vyd/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' (no such file), '/System/Volumes/
Preboot/Cryptexes/OS/private/tmp/glib-20241104-31967-kl1vyd/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' (no 
such file), '/private/tmp/glib-20241104-31967-kl1vyd/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' (no such file)
Traceback (most recent call last):
  File "/Users/arjan/Development/glib/./test-glib.py", line 4, in <module>
    from gi.repository import GIRepository
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
[...]
```

When built from source the absolute path is indeed in `GLib-2.0.typelib`. This is not the case with the bottle 😕 .

Reply:
Try applying this patch and see if it helps:
```diff
diff --git a/Formula/g/glib.rb b/Formula/g/glib.rb
index 6accd6db74e2..61a76802993e 100644
--- a/Formula/g/glib.rb
+++ b/Formula/g/glib.rb
@@ -50,6 +50,14 @@ class Glib < Formula
   resource "gobject-introspection" do
     url "https://download.gnome.org/sources/gobject-introspection/1.82/gobject-introspection-1.82.0.tar.xz"
     sha256 "0f5a4c1908424bf26bc41e9361168c363685080fbdb87a196c891c8401ca2f09"
+
+    # Fix library search path on non-/usr/local installs (e.g. Apple Silicon)
+    # See: https://github.com/Homebrew/homebrew-core/issues/75020
+    #      https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/273
+    patch do
+      url "https://gitlab.gnome.org/tschoonj/gobject-introspection/-/commit/a7be304478b25271166cd92d110f251a8742d16b.diff"
+      sha256 "740c9fba499b1491689b0b1216f9e693e5cb35c9a8565df4314341122ce12f81"
+    end
   end
 
   # replace several hardcoded paths with homebrew counterparts
```

Please test it out locally first.

Reply:
This patch doesn't help. It achieves the same as the proposed patch, but via meson build options.

The problem I'm facing, as outlines in #197684, is that the typelibs are compiled wrongly, and refer to the "staging" build artifacts instead of the installed libraries. This results in messages like this:

```
** (process:54915): WARNING **: 13:30:17.245: Failed to load shared library '/private/tmp/glib-
20241114-44813-gq42du/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' referenced by the 
typelib: dlopen(/private/tmp/glib-20241114-44813-gq42du/glib-2.82.2/staging/lib/libgobject-
2.0.0.dylib, 0x0009): tried: '/opt/gtk/lib/libgobject-2.0.0.dylib' (no such file), '/private/tmp/
glib-20241114-44813-gq42du/glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' (no such file), 
'/System/Volumes/Preboot/Cryptexes/OS/private/tmp/glib-20241114-44813-gq42du/glib-2.82.2/
staging/lib/libgobject-2.0.0.dylib' (no such file), '/private/tmp/glib-20241114-44813-gq42du/
glib-2.82.2/staging/lib/libgobject-2.0.0.dylib' (no such file)
```

You can see this issue also in the output of the failing jobs.

What confuses me is that the bottles have it right, though 😕